components/UpdateBlogSection/DangerZone.tsx

```diff
@@ -0,0 +1,53 @@
+import { useRouter } from 'next/router';
+
+import { useMutation } from '../../hooks/useMutation';
+import { deleteBlog } from '../../utils/api/deleteBlog';
+import { Button } from '../Button/Button';
+
+import styles from './dangerZone.module.scss';
+
+type DangerZoneProps = {
+  readonly blogId: string;
+};
+
+export const DangerZone = ({ blogId }: DangerZoneProps) => {
+  const deleteMutation = useMutation(deleteBlog);
+  const router = useRouter();
+
+  async function handleDelete() {
+    if (
+      window.confirm(
+        'Czy na pewno chcesz usunąć bloga? Wszystkie jego dane oraz wpisy zostaną nieodwracalnie usunięte.',
+      )
+    ) {
+      try {
+        await deleteMutation.mutate(blogId);
+        await router.push('/admin');
```

Comment:
Czyli nawet jak się nie uda usunąć bloga to nas przenosi na `/admin`? Raczej mało intuicyjne. Powinno przenieść dopiero po udanym usunięciu.

Reply:
Przecież jak się nie uda to wyrzuci błąd i router.push już się nie wywoła 👀

Reply:
Fetch nie rzuca błędu jak dostanie jakiś response, choćby `404`.

Reply:
A nasz fetcher tego nie robi?

Reply:
I useMutation?

Reply:
A, nie robi, racja